javascript/ecmascript-2015/function-improvements/arrow-functions-part-2.md

```diff
@@ -89,57 +89,6 @@ This means that in the `countUp()` function, `this` will be the global object wh
 
 On the other hand, arrow functions always treat `this` like a variable from the upper scope, no matter how we call them. This leads to the `countDown()` function modifying the `counter` property on the instance of `Counter`.
```

Comment:
Now that I saw how the insights look when split, I think it would be a better idea to add `arrow-functions-part-2` content to a footnote here. it is too reliant on this insight to make it individual.

what do you think @Stefan-Stojanovic @nem035 ?

```suggestion
On the other hand, arrow functions always treat `this` like a variable from the upper scope, no matter how we call them. This leads to the `countDown()` function modifying the `counter` property on the instance of `Counter`. If you're still unsure on how this works, we've added some comments to the example in this footnote[1].
```

``````
[1:Commented Code]
Here's the code snippet again with the explanations:
```js
function Counter() {
  // We wrote `Counter` in such
  // a way that it expects `this`
  // to be an instance of itself.
  // In other words, we expect
  // `Counter` to be called using the
  // keyword `new`, which will initialize
  // `this` to be an instance object, and
  // we assign a `counter` variable to
  // that object
  this.counter = 0;

  function countUp() {
    // This function will have `this`
    // as the global object because setInterval
    // will call it like a regular function,
    // i.e. fn()

    // this means that in here
    // we increment a `counter`
    // on the global object,
    // not the one on `Counter`
    this.counter++;
  }

  const countDown = () => {
    // this arrow function
    // will look up `this`
    // in the scope above, which is the
    // scope of Counter, and use that.
    // In here, we update the `counter`
    // on whatever `this` points to 
    // within `Counter`, which in our 
    // case should be an instance of
    // `Counter`
    this.counter--;
  }

  setInterval(countUp, 1000);
  setInterval(countDown, 1000)
}

// the behavior explained above
// is correct because we are
// calling `Counter` with `new`
let count = new Counter();
```
``````

Reply:
Sure, that works